plugins/pulp_docker/plugins/distributors/publish_steps.py

```diff
@@ -91,19 +94,30 @@ def initialize(self):
                                                     self.get_repo())
         self.redirect_context.initialize()
 
-    def process_unit(self, unit):
+    def get_count(self):
+        repo = self.get_repo()
+        return repo.content_unit_counts.get(constants.IMAGE_TYPE_ID, 0)
+
+    def get_iterator(self):
+        for content_unit in repo_controller.find_repo_content_units(
```

Comment:
It looks like the ``find_repo_content_units`` method already returns a generator. Rather than make a new one with the loop and yield statement, you can just return the original.